packages/node/src/indexer/api.service.ts

```diff
@@ -360,12 +345,30 @@ export class ApiService implements OnApplicationShutdown {
   ): StorageKey[] {
     return calls.map((callMultiArg) => {
       if (callMultiArg instanceof Array) {
-        const [storageFunc, ...args] = callMultiArg;
-        const key = new StorageKey(api.registry, storageFunc.key(...args));
+        let key: StorageKey;
+        const storageFunc = callMultiArg[0];
+        const args = callMultiArg.slice(1);
+        if (storageFunc.creator.meta.type.isPlain) {
+          key = new StorageKey(api.registry, storageFunc.creator);
+        } else {
+          if (storageFunc.creator.meta.type.asMap.hashers.length === 1) {
+            // single arg
+            key = new StorageKey(api.registry, storageFunc.key(args[0]));
+          } else {
+            // can be array or expanded
+            // eg. args = [call,[2038, `HAGcVQikZmEEgBBaChwjTVdwdA53Qopg2AYUtqw738C5kUq`]]
+            if (args.length === 1 && args[0] instanceof Array) {
```

Comment:
Seems quit4e dangerous to assume you can extend args by checking the args type. What if it is meant to not be expanded but the args happens to be an array. There're plenty types in polkadot-js extend Array, like `Tuple`